mne/forward/_lead_dots.py

```diff
@@ -174,15 +183,19 @@ def _comp_sums_meg(beta, ctheta, lut_fun, n_fact, volume_integral):
     #  * sums[:, 1]    n/(2n+1) beta^(n+1) P_n'
     #  * sums[:, 2]    n/((2n+1)(n+1)) beta^(n+1) P_n'
     #  * sums[:, 3]    n/((2n+1)(n+1)) beta^(n+1) P_n''
-    coeffs = lut_fun(ctheta)
-    bbeta = np.cumprod(np.tile(beta[np.newaxis], (n_fact.shape[0], 1)),
-                       axis=0)
+    bbeta = np.tile(beta[np.newaxis], (n_fact.shape[0], 1))
+    bbeta = np.cumprod(bbeta, axis=0, out=bbeta)  # run inplace
     bbeta *= beta
     # This is equivalent, but slower:
     # sums = np.sum(bbeta[:, :, np.newaxis].T * n_fact * coeffs, axis=1)
     # sums = np.rollaxis(sums, 2)
-    sums = np.einsum('ji,jk,ijk->ki', bbeta, n_fact, coeffs)
-    return sums
+    # or
+    # sums = np.einsum('ji,jk,ijk->ki', bbeta, n_fact, lut_fun(ctheta)))
+    sums = list()
+    for k in range(len(n_fact[0])):  # lookup in blocks to save memory
+        sums.append(np.einsum('ji,j,ij->i', bbeta, n_fact[:, k],
+                              lut_fun(ctheta, block=k)))
+    return np.array(sums)
```

Comment:
I meant this one: http://mne-tools.github.io/stable/auto_examples/preprocessing/plot_virtual_evoked.html?highlight=as_type by why not that one too